sklearn/metrics/ranking.py

```diff
@@ -624,3 +625,78 @@ def coverage_error(y_true, y_score, sample_weight=None):
     coverage = coverage.filled(0)
 
     return np.average(coverage, weights=sample_weight)
+
+
+def label_ranking_loss(y_true, y_score, sample_weight=None):
+    """Compute Ranking loss measure
+
+    Compute the average number of labels pairs that are incorrectly ordered
+    given y_score weighted by the size of the label set and the number of
+    labels not in the labels set.
+
+    This is similar to the error set size, but weighted by the number of
+    relevant and irrelevant labels. The best performance is achieved with
+    a ranking loss of zero.
+
+    Parameters
+    ----------
+    y_true : array or sparse matrix, shape = [n_samples, n_labels]
+        True binary labels in binary indicator format.
+
+    y_score : array, shape = [n_samples, n_labels]
+        Target scores, can either be probability estimates of the positive
+        class, confidence values, or binary decisions.
+
+    sample_weight : array-like of shape = [n_samples], optional
+        Sample weights.
+
+    Return
+    ------
+    loss : float
+
+    References
+    ----------
+    .. [1] Tsoumakas, G., Katakis, I., & Vlahavas, I. (2010).
+           Mining multi-label data. In Data mining and knowledge discovery
+           handbook (pp. 667-685). Springer US.
+
+    """
+    y_true = check_array(y_true, ensure_2d=False, accept_sparse='csr')
+    y_score = check_array(y_score, ensure_2d=False)
+    check_consistent_length(y_true, y_score, sample_weight)
+
+    y_type = type_of_target(y_true)
+    if y_type not in ("multilabel-indicator",):
+        raise ValueError("{0} format is not supported".format(y_type))
+
+    if y_true.shape != y_score.shape:
+        raise ValueError("y_true and y_score have different shape")
+
+    n_samples, n_labels = y_true.shape
+
+    y_true = csr_matrix(y_true)
+
+    loss = np.zeros(n_samples)
+    for i, (start, stop) in enumerate(zip(y_true.indptr, y_true.indptr[1:])):
+        unique_scores, unique_inverse = np.unique(y_score[i],
+                                                  return_inverse=True)
+        true_at_reversed_rank = np.bincount(
+            unique_inverse[y_true.indices[start:stop]],
+            minlength=unique_scores.size)
+        all_at_reversed_rank = np.bincount(unique_inverse,
+                                           minlength=unique_scores.size)
+        false_at_reversed_rank = all_at_reversed_rank - true_at_reversed_rank
+
+        loss[i] = np.dot(true_at_reversed_rank.cumsum(),
+                         false_at_reversed_rank)
```

Comment:
I have to admit that I have a hard time understanding what is going here in this loop. Given the tests, I assume this gets the job done correctly, but it may be nice to add some inline comments to make it easier to read.